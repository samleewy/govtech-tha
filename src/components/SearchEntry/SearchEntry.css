.title a {
    color: #1C76D5;
    font-weight: 600;
    text-decoration: none;
}

.title .highlight {
    color: #005bba;
    font-weight: 700;
}

.desc {
    margin-top: 5px;
    color: #424242;
    font-size: 14px;
    font-weight: 400;
}

.desc .highlight {
    color: #212121;
    font-weight: 700;
}

.link {
    margin-top: 5px;
    font-size: 12px;
}

.link a {
    color: #686868;
    text-decoration: none;
}

.link a:hover {
    text-decoration: underline;
}