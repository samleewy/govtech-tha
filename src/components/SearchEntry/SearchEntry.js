import React from 'react' 
import './SearchEntry.css'

const SearchEntry = ({ title, desc, link }) => {
    return (
        <div>
            <div class='title'>
                <a href={link}>{ title }</a>
            </div>
            <div class='desc'>
                1 Sep 2021 â€” Plan Your <span class='highlight'>Finances</span> Applying for a Flat From 
                HDB Before You Apply What to Expect Steps Involved Buying an HDB Resale Flat Eligibility 
                Criteria What to Expect...
            </div>
            <div class='link'>
                { link }
            </div>
        </div>
    )
}

export default SearchEntry