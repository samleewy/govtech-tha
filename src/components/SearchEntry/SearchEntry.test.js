import { render } from "@testing-library/react";
import SearchEntry from "./SearchEntry";

describe("This suit is to test SearchEntry component", () => {
    const testComponent = <SearchEntry title="Test Title" desc="Test Description" link="www.testlink.com" />
    it('should render correctly', () => {
        const { asFragment } = render(testComponent)
        expect(asFragment()).toMatchSnapshot();
    })
})