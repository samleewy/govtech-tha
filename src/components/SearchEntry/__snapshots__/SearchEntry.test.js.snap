// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test SearchEntry component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="title"
    >
      <a
        href="www.testlink.com"
      >
        Test Title
      </a>
    </div>
    <div
      class="desc"
    >
      1 Sep 2021 â€” Test Description
    </div>
    <div
      class="link"
    >
      <a
        href="www.testlink.com"
      >
        www.testlink.com
      </a>
    </div>
  </div>
</DocumentFragment>
`;
